site/docs/tools/client_config.md

```diff
@@ -125,6 +126,82 @@ This is also the default authentication type if nothing else is configured.
 For the `BEARER` Authentication Type the `authentication.token` property should be set to a valid
 [OpenID token](https://openid.net/specs/openid-connect-core-1_0.html).
 
+This authentication type is recommended only when the issued access token has a lifespan large
+enough to cover the duration of the entire Nessie client's session. Once the token is expired, the
+Nessie client will not be able to refresh it and will have to be restarted, with a different token.
+If the token needs to be refreshed periodically, then the `OAUTH2` authentication type should be
+preferred to this one.
+
+## Authentication Type `OAUTH2`
+
+The `OAUTH2` Authentication Type is able to authenticate against an OAuth2 server and obtain a valid
+access token. Only Bearer access tokens are currently supported. The access token is then used to
+authenticate against Nessie. The client will automatically refresh the access token. This
+authentication type is recommended when the access token has a lifespan shorter than the Nessie
+client's session lifespan.
+
+Note that the Nessie server must be configured to accept OAuth2 tokens from the same server. For
+example, if the OAuth2 server is Keycloak, this can be done by defining the following properties in
+the `application.properties` file of the Nessie server:
+
+```properties
+nessie.server.authentication.enabled=true
+quarkus.oidc.auth-server-url=https://<keycloak-server>/realms/<realm-name>
+```
+
+The following properties are available for the `OAUTH2` authentication type:
+
+* `authentication.oauth2.token-endpoint`: the URL of the OAuth2 token endpoint; this should include
+  not only the OAuth2 server's address, but also the path to the token REST resource, if any. For
+  Keycloak, this is typically
+  `https://<keycloak-server>/realms/<realm-name>/protocol/openid-connect/token`. Required.
+
+* `authentication.oauth2.grant-type`: the grant type to use when authenticating against the OAuth2
+  server. Valid values are: `client_credentials` or `password`. Optional, defaults to
+  `client_credentials`. For both grant types, a client ID and secret must be provided and will be
+  used to authenticate the client against the OAuth2 server. When using the "password" grant type, a
+  username and password must also be provided, and are used to authenticate the user. Both client
+  and user must be properly configured with appropriate permissions in the OAuth2 server for the
+  authentication to succeed.
+
+* `authentication.oauth2.client-id`: the client ID to use when authenticating against the OAuth2
+  server. Required.
+
+* `authentication.oauth2.client-secret`: the client secret to use when authenticating against the
+  OAuth2 server. Required.
+
+* `authentication.oauth2.username`: the username to use when authenticating against the OAuth2
```

Comment:
@dimas-b I also wanted to mention this: I created 2 new options for username/password in the context of OAuth2, rather than reusing the 2 existing (deprecated) settings: `authentication.username` and `authentication.password`.

Are we OK with this?

Reply:
Yes, these are fine, I think because they are specific to the (standard) OAuth2 protocol. The deprecated `authentication.password` comes from the time when Nessie itself was validating passwords.

Reply:
It might be ok to remove `authentication.password` and `authentication.username` now, I guess :)